### ISSUE TITLE
unreachable false positive with enums

### ISSUE THREAD

**Issue by sk-** [python/mypy#18808]
**Bug Report**
When using enums that extend both from `Enum` and `str` equality checks are marked as impossible hence producing unreachable warnings, however the code is reachable.

**To Reproduce**
[Playground](https://mypy-play.net/?mypy=latest&python=3.12&flags=strict-equality%2Cwarn-unreachable&gist=67b27e680fc6ab746ad6b463d7993d75)
```python
from enum import Enum, StrEnum
from typing import TYPE_CHECKING, reveal_type

class EnumMulti(str, Enum):
    label = "label"


def test_str_enums_values() -> None:
    if TYPE_CHECKING:
        reveal_type(EnumMulti.label)
    assert EnumMulti.label == "label"
    print("end")
    
test_str_enums_values()
```

**Expected Behavior**
Code is not marked as unreachable because when run the code actually prints the string `end`.

**Actual Behavior**

```
main.py:10: note: Revealed type is "Literal[__main__.EnumMulti.label]?"
main.py:12: error: Statement is unreachable  [unreachable]
Found 1 error in 1 file (checked 1 source file)
```

This is strange because no other warning/error is raised. I would have expected at least to get a `comparison-overlap` or `redundant-expr`. In fact, if we change the comparison to `assert EnumMulti.label == 2` we get the `comparison-overlap` error and the `unreachable` error, but if we change the comparison to `assert "foo" == 2` we only get the `comparison-overlap` error.

**Your Environment**

- Mypy version used: mypy 1.15.0 (compiled: yes)
- Mypy command-line flags: --strict-equality --warn-unreachable
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.12.9

**Comment by A5rocks** [comment]
Duplicate of #17162
